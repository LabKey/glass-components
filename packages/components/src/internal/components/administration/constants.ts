import { PermissionRoles } from '@labkey/api';
import { Map } from 'immutable';

const ROLE_ACTION_DESCRIPTION =
    'and creating and editing sample types, source types, assay designs, and job templates.';

export const SECURITY_ROLE_DESCRIPTIONS = Map<string, string>([
    [
        PermissionRoles.ApplicationAdmin,
        'Application Administrators have full control over the application. This includes user management, permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.ProjectAdmin,
        'Project Administrators have full control over the application in this project container. This includes user management, permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.FolderAdmin,
        'Folder Administrators have control over the folder specific pieces of the application. This includes permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.Editor,
        'Editors may add new information or edit data related to samples, sources, assays, jobs, but not storage locations.',
    ],
    [PermissionRoles.Reader, 'Readers have a read-only view of the application.'],
]);

export const HOSTED_APPLICATION_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.Editor, 'Editor'],
    [PermissionRoles.EditorWithoutDelete, 'Editor without Delete'],
    [PermissionRoles.Reader, 'Reader'],
]);

export const SITE_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.ApplicationAdmin, 'Application Administrator'],
]);

export const APPLICATION_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.ProjectAdmin, 'Project Administrator'],
    [PermissionRoles.FolderAdmin, 'Folder Administrator'],
    [PermissionRoles.Editor, 'Editor'],
    [PermissionRoles.EditorWithoutDelete, 'Editor without Delete'],
    [PermissionRoles.Reader, 'Reader'],
]);


