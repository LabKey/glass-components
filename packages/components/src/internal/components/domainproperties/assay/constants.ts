export const PROPERTIES_HEADER_ID = 'assay-properties-hdr';
export const FORM_ID_PREFIX = 'assay-design-';

export const FORM_IDS = {
    ASSAY_NAME: FORM_ID_PREFIX + 'name',
    ASSAY_DESCRIPTION: FORM_ID_PREFIX + 'description',
    AUTO_LINK_TARGET: FORM_ID_PREFIX + 'autoCopyTargetContainerId',
    AUTO_LINK_CATEGORY: FORM_ID_PREFIX + 'autoLinkCategory',
    BACKGROUND_UPLOAD: FORM_ID_PREFIX + 'backgroundUpload',
    DETECTION_METHOD: FORM_ID_PREFIX + 'selectedDetectionMethod',
    EDITABLE_RUNS: FORM_ID_PREFIX + 'editableRuns',
    EDITABLE_RESULTS: FORM_ID_PREFIX + 'editableResults',
    METADATA_INPUT_FORMAT: FORM_ID_PREFIX + 'selectedMetadataInputFormat',
    PLATE_TEMPLATE: FORM_ID_PREFIX + 'selectedPlateTemplate',
    PROTOCOL_TRANSFORM_SCRIPTS: FORM_ID_PREFIX + 'protocolTransformScripts',
    QC_ENABLED: FORM_ID_PREFIX + 'qcEnabled',
    SAVE_SCRIPT_FILES: FORM_ID_PREFIX + 'saveScriptFiles',
    PLATE_METADATA: FORM_ID_PREFIX + 'plateMetadata',
    STATUS: FORM_ID_PREFIX + 'status',
};

export const BOOLEAN_FIELDS = [
    FORM_IDS.BACKGROUND_UPLOAD,
    FORM_IDS.EDITABLE_RUNS,
    FORM_IDS.EDITABLE_RESULTS,
    FORM_IDS.QC_ENABLED,
    FORM_IDS.SAVE_SCRIPT_FILES,
    FORM_IDS.PLATE_METADATA,
    FORM_IDS.STATUS,
];
