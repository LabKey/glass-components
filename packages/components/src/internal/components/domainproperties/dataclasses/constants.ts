export const DATACLASS_DOMAIN_SYSTEM_FIELDS = [
    {
        Name: 'Name',
        Label: 'Name',
        'Data Type': 'Text',
        Required: false,
        Description: 'Contains a short name for this data object.',
    },
    {
        Name: 'Description',
        Label: 'Description',
        'Data Type': 'Text',
        Required: false,
        Description: 'Contains a description for this data object',
    },
];
