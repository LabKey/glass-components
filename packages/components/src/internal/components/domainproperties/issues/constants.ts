export const ISSUES_LIST_DEF_SINGULAR_PLURAL_TIP =
    'The Singular item name and Plural item name fields control the display ' +
    'name for an "issue" across the Issues module. For example, you might instead refer to issues as "Bugs", "Tickets", or "Opportunities" depending on the context.';

export const ISSUES_LIST_DEF_SORT_DIRECTION_TIP =
    'By default, comments on an issue are shown in the order they are added, oldest first. Change the Comment Sort Direction to newest first if you prefer.';
export const ISSUES_LIST_GROUP_ASSIGN_TIP =
    "Select a specific group from the dropdown, both site and project level groups are available. If left blank, the 'Assigned To' field will display all users who are a) assigned the role of Editor or higher in the folder, and b) are members of at least one project-level security group.";
export const ISSUES_LIST_USER_ASSIGN_TIP =
    'In some workflows it is useful to have a default user to whom all issues are assigned, such as for initial triage and balancing assignments across a group. If left blank, no user will be assigned by default.';
export const ISSUES_LIST_RELATED_FOLDER_TIP =
    'A related issue can be created in any issue list that the user can insert into. This setting controls the default folder selection in the related issue dialog box.';
export const ISSUES_LIST_RESTRICTED_TRACKER_TIP =
    'A restricted issue list will limit viewing and updating of all issues in the list. When the checkbox is selected only users who are either assigned to the issue or on the notify list ' +
    'will have access to an issue. An additional group can optionally be configured to provide access to any members of that group.';
export const ISSUES_LIST_RESTRICTED_GROUP_TIP =
    'An optional project or site group that can give members of the group access to the issues in a restricted issue list.';
