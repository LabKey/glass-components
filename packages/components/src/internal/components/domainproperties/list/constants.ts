export const DATA_INDEXING_TIP =
    'Not recommend for large lists with frequent updates, since updating any item will cause re-indexing of the entire list.';
export const DOCUMENT_TITLE_TIP =
    'Any text you want displayed and indexed as the search result title (ex. ListName - ${Key} ${value}). Leave the document title blank to use the default title.';
export const CUSTOM_TEMPLATE_TIP = 'Example: ${Key} ${value}';

export const DISCUSSION_LINKS_TIP =
    'Optionally allow one or more discussion links to be shown on the details view of each list item.';
export const SEARCH_INDEXING_TIP = 'Controls how this list is indexed for search within LabKey Server.';
