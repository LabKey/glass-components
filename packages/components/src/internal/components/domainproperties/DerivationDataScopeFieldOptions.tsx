import React from 'react';
import { Checkbox, Col, Row } from 'react-bootstrap';

import { createFormInputId, createFormInputName, getCheckedValue } from './actions';
import { isFieldFullyLocked } from './propertiesUtil';
import { DOMAIN_FIELD_DERIVATION_DATA_SCOPE } from './constants';
import { IDerivationDataScope, ITypeDependentProps } from './models';
import { SectionHeading } from './SectionHeading';
import { LabelHelpTip } from "../../..";

interface Props extends ITypeDependentProps {
    value?: string;
    config?: IDerivationDataScope;
}

const CHILD_ONLY = 'ChildOnly';

export class DerivationDataScopeFieldOptions extends React.PureComponent<Props, any> {

    static defaultProps = {
        config: {
            show: true,
            disable: false,
            fieldLabel: 'Derivation Data Scope'
        },
    };

    handleCheckboxChange = (evt) => {
        const { onChange } = this.props;

        if (onChange) {
            const value = getCheckedValue(evt);
            onChange(evt.target.id, value ? CHILD_ONLY : '');
        }

    };

    renderHelp() {
        const { config } = this.props;
        return config.helpLinkNode ? config.helpLinkNode :
            <LabelHelpTip title="Child specific field">
                <div>
                    <p>
                        If checked, this field is only available for child data.
                    </p>
                </div>
            </LabelHelpTip>
    }

    render() {
        const { index, label, value, lockType, domainIndex, config } = this.props;

        return (
            <div>
                <Row>
                    <Col xs={12}>
                        <SectionHeading title={label} cls="domain-field-section-hdr" />
                    </Col>
                </Row>
                <Row>
                    <Col xs={3}>
                        <Checkbox
                            checked={value?.toLocaleLowerCase() === CHILD_ONLY.toLocaleLowerCase()}
                            onChange={this.handleCheckboxChange}
                            id={createFormInputId(DOMAIN_FIELD_DERIVATION_DATA_SCOPE, domainIndex, index)}
                            disabled={config.disable || isFieldFullyLocked(lockType)}
                            name={createFormInputName(DOMAIN_FIELD_DERIVATION_DATA_SCOPE)}
                        >
                            {config.fieldLabel}
                            {this.renderHelp()}
                        </Checkbox>
                    </Col>
                </Row>
            </div>
        );
    }
}
