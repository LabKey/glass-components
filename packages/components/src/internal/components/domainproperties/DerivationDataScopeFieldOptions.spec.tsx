import { mount } from 'enzyme';
import React from 'react';

import { Radio } from 'react-bootstrap';

import { Alert } from '../base/Alert';

import { DERIVATION_DATA_SCOPES, DOMAIN_FIELD_NOT_LOCKED } from './constants';
import { DerivationDataScopeFieldOptions } from './DerivationDataScopeFieldOptions';
import { PropDescType, TEXT_TYPE, UNIQUE_ID_TYPE } from "./PropDescType";

describe('DerivationDataScopeFieldOptions', () => {
    test('Default config, new field', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: false,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const sectionLabel = wrapper.find({ className: 'domain-field-section-heading domain-field-section-hdr' });
        expect(sectionLabel.length).toEqual(1);
        expect(sectionLabel.text()).toEqual('Derivation Data Scope');

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);
        expect(radios.at(0).prop('checked')).toBeTruthy();
        expect(radios.at(0).prop('disabled')).toBeFalsy();
        expect(radios.at(0).text()).toBe('Editable for parent data only (default)');
        expect(radios.at(1).prop('checked')).toBeFalsy();
        expect(radios.at(1).prop('disabled')).toBeFalsy();
        expect(radios.at(1).text()).toBe('Editable for child data only');
        expect(radios.at(2).prop('checked')).toBeFalsy();
        expect(radios.at(2).prop('disabled')).toBeFalsy();
        expect(radios.at(2).text()).toBe('Editable for parent and child data independently');
        expect(wrapper.find(Alert)).toHaveLength(0);

        wrapper.unmount();
    });

    test('Existing field, value = ParentOnly', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: true,
            value: DERIVATION_DATA_SCOPES.PARENT_ONLY,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);
        expect(radios.at(0).prop('checked')).toBeTruthy();
        expect(radios.at(0).prop('disabled')).toBeFalsy();
        expect(radios.at(1).prop('checked')).toBeFalsy();
        expect(radios.at(1).prop('disabled')).toBeTruthy();
        expect(radios.at(2).prop('checked')).toBeFalsy();
        expect(radios.at(2).prop('disabled')).toBeFalsy();
        expect(wrapper.find(Alert)).toHaveLength(0);

        wrapper.unmount();
    });

    test('Existing field, value is empty', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: true,
            value: '',
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);
        expect(radios.at(0).prop('checked')).toBeTruthy();
        expect(radios.at(0).prop('disabled')).toBeFalsy();
        expect(radios.at(1).prop('checked')).toBeFalsy();
        expect(radios.at(1).prop('disabled')).toBeTruthy();
        expect(radios.at(2).prop('checked')).toBeFalsy();
        expect(radios.at(2).prop('disabled')).toBeFalsy();
        expect(wrapper.find(Alert)).toHaveLength(0);

        wrapper.unmount();
    });

    test('Existing field, value = ChildOnly', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: true,
            value: DERIVATION_DATA_SCOPES.CHILD_ONLY,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);
        expect(radios.at(0).prop('checked')).toBeFalsy();
        expect(radios.at(0).prop('disabled')).toBeTruthy();
        expect(radios.at(0).text()).toBe('Editable for parent data only (default)');
        expect(radios.at(1).prop('checked')).toBeTruthy();
        expect(radios.at(1).prop('disabled')).toBeFalsy();
        expect(radios.at(1).text()).toBe('Editable for child data only');
        expect(radios.at(2).prop('checked')).toBeFalsy();
        expect(radios.at(2).prop('disabled')).toBeFalsy();
        expect(radios.at(2).text()).toBe('Editable for parent and child data independently');
        expect(wrapper.find(Alert)).toHaveLength(0);

        wrapper.unmount();
    });

    test('Existing field, value = All', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: true,
            value: DERIVATION_DATA_SCOPES.ALL,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);
        expect(radios.at(0).prop('checked')).toBeFalsy();
        expect(radios.at(0).prop('disabled')).toBeTruthy();
        expect(radios.at(1).prop('checked')).toBeFalsy();
        expect(radios.at(1).prop('disabled')).toBeTruthy();
        expect(radios.at(2).prop('checked')).toBeTruthy();
        expect(radios.at(2).prop('disabled')).toBeFalsy();
        expect(wrapper.find(Alert)).toHaveLength(0);

        wrapper.unmount();
    });

    test('With config, show = true', () => {
        const label = 'Sample/Aliquot Options';
        const warning =
            "Updating a 'Samples Only' field to be 'Samples and Aliquots' will blank out the field values for all aliquots. This action cannot be undone.";
        const props = {
            index: 1,
            domainIndex: 1,
            label,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            config: {
                show: true,
                sectionTitle: 'Sample/Aliquot Options',
                labelAll: 'Separately editable for samples and aliquots',
                labelChild: 'Editable for aliquots only',
                labelParent: 'Editable for samples only (default)',
                helpLinkNode: <>help</>,
                scopeChangeWarning: warning,
            },
            isExistingField: true,
            value: DERIVATION_DATA_SCOPES.PARENT_ONLY,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const sectionLabel = wrapper.find({ className: 'domain-field-section-heading domain-field-section-hdr' });
        expect(sectionLabel.length).toEqual(1);
        expect(sectionLabel.text()).toEqual(label);

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);
        expect(radios.at(0).prop('checked')).toBeTruthy();
        expect(radios.at(0).text()).toBe('Editable for samples only (default)');
        expect(radios.at(1).prop('checked')).toBeFalsy();
        expect(radios.at(1).text()).toBe('Editable for aliquots only');
        expect(radios.at(2).prop('checked')).toBeFalsy();
        expect(radios.at(2).text()).toBe('Separately editable for samples and aliquots');
        expect(wrapper.find(Alert)).toHaveLength(0);

        wrapper.unmount();
    });

    test('With config, show = false', () => {
        const label = 'Sample/Aliquot Options';
        const props = {
            index: 1,
            domainIndex: 1,
            label,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            config: {
                show: false,
            },
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);
        expect(wrapper).toEqual({});
        wrapper.unmount();
    });

    test('With config, isExistingField is not applicable', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: true,
            value: DERIVATION_DATA_SCOPES.ALL,
            config: {
                show: true,
                dataTypeFilter: (dataType: PropDescType) => !dataType.isUniqueId(),
            },
            fieldDataType: UNIQUE_ID_TYPE,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        expect(wrapper).toEqual({});

        wrapper.unmount();
    });

    test('With config, isExistingField is applicable', () => {
        const props = {
            index: 1,
            domainIndex: 1,
            label: null,
            onChange: jest.fn(),
            lockType: DOMAIN_FIELD_NOT_LOCKED,
            isExistingField: true,
            value: DERIVATION_DATA_SCOPES.ALL,
            config: {
                show: true,
                dataTypeFilter: (dataType: PropDescType) => !dataType.isUniqueId(),
            },
            fieldDataType: TEXT_TYPE,
        };

        const wrapper = mount(<DerivationDataScopeFieldOptions {...props} />);

        const radios = wrapper.find(Radio);
        expect(radios).toHaveLength(3);

        wrapper.unmount();
    });
});
