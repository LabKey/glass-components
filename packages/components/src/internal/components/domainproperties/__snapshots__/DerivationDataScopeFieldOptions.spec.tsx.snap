// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DerivationDataScopeFieldOptions Derivation Data Scope Field Options 1`] = `
<DerivationDataScopeFieldOptions
  config={
    Object {
      "disable": false,
      "fieldLabel": "Derivation Data Scope",
      "show": true,
    }
  }
  domainIndex={1}
  index={1}
  label="Aliquot properties"
  lockType="NotLocked"
  onChange={[MockFunction]}
  value="ChildOnly"
>
  <div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <div
        className="row"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}
        >
          <div
            className="col-xs-12"
          >
            <SectionHeading
              cls="domain-field-section-hdr"
              title="Aliquot properties"
            >
              <div
                className="domain-field-section-heading domain-field-section-hdr"
              >
                Aliquot properties
              </div>
            </SectionHeading>
          </div>
        </Col>
      </div>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <div
        className="row"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={3}
        >
          <div
            className="col-xs-3"
          >
            <Checkbox
              bsClass="checkbox"
              checked={true}
              disabled={false}
              id="domainpropertiesrow-derivationDataScope-1-1"
              inline={false}
              name="domainpropertiesrow-derivationDataScope"
              onChange={[Function]}
              title=""
            >
              <div
                className="checkbox"
              >
                <label
                  title=""
                >
                  <input
                    checked={true}
                    disabled={false}
                    id="domainpropertiesrow-derivationDataScope-1-1"
                    name="domainpropertiesrow-derivationDataScope"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Derivation Data Scope
                  <Component
                    customStyle={Object {}}
                    id="tooltip"
                    size="1x"
                    title="Child specific field"
                  >
                    <span
                      className="label-help-target"
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                    >
                      <FontAwesomeIcon
                        border={false}
                        className="label-help-icon"
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              512,
                              512,
                              Array [],
                              "f059",
                              "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z",
                            ],
                            "iconName": "question-circle",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size="1x"
                        spin={false}
                        style={Object {}}
                        swapOpacity={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-question-circle fa-w-16 fa-1x label-help-icon"
                          data-icon="question-circle"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                      <Overlay
                        animation={[Function]}
                        placement="right"
                        rootClose={false}
                        show={false}
                        target={
                          <span
                            class="label-help-target"
                          >
                            <svg
                              aria-hidden="true"
                              class="svg-inline--fa fa-question-circle fa-w-16 fa-1x label-help-icon"
                              data-icon="question-circle"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        }
                      >
                        <Overlay
                          placement="right"
                          rootClose={false}
                          show={false}
                          target={
                            <span
                              class="label-help-target"
                            >
                              <svg
                                aria-hidden="true"
                                class="svg-inline--fa fa-question-circle fa-w-16 fa-1x label-help-icon"
                                data-icon="question-circle"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                viewBox="0 0 512 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          }
                          transition={[Function]}
                        />
                      </Overlay>
                    </span>
                  </Component>
                </label>
              </div>
            </Checkbox>
          </div>
        </Col>
      </div>
    </Row>
  </div>
</DerivationDataScopeFieldOptions>
`;

exports[`DerivationDataScopeFieldOptions With config 1`] = `
<DerivationDataScopeFieldOptions
  config={
    Object {
      "disable": false,
      "fieldLabel": "Aliquot Only",
      "sectionTitle": "Aliquot field",
      "show": true,
    }
  }
  domainIndex={1}
  index={1}
  label="Aliquot field"
  lockType="NotLocked"
  onChange={[MockFunction]}
>
  <div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <div
        className="row"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}
        >
          <div
            className="col-xs-12"
          >
            <SectionHeading
              cls="domain-field-section-hdr"
              title="Aliquot field"
            >
              <div
                className="domain-field-section-heading domain-field-section-hdr"
              >
                Aliquot field
              </div>
            </SectionHeading>
          </div>
        </Col>
      </div>
    </Row>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <div
        className="row"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={3}
        >
          <div
            className="col-xs-3"
          >
            <Checkbox
              bsClass="checkbox"
              checked={false}
              disabled={false}
              id="domainpropertiesrow-derivationDataScope-1-1"
              inline={false}
              name="domainpropertiesrow-derivationDataScope"
              onChange={[Function]}
              title=""
            >
              <div
                className="checkbox"
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={false}
                    id="domainpropertiesrow-derivationDataScope-1-1"
                    name="domainpropertiesrow-derivationDataScope"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Aliquot Only
                  <Component
                    customStyle={Object {}}
                    id="tooltip"
                    size="1x"
                    title="Child specific field"
                  >
                    <span
                      className="label-help-target"
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                    >
                      <FontAwesomeIcon
                        border={false}
                        className="label-help-icon"
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              512,
                              512,
                              Array [],
                              "f059",
                              "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z",
                            ],
                            "iconName": "question-circle",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size="1x"
                        spin={false}
                        style={Object {}}
                        swapOpacity={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-question-circle fa-w-16 fa-1x label-help-icon"
                          data-icon="question-circle"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                      <Overlay
                        animation={[Function]}
                        placement="right"
                        rootClose={false}
                        show={false}
                      >
                        <Overlay
                          placement="right"
                          rootClose={false}
                          show={false}
                          transition={[Function]}
                        />
                      </Overlay>
                    </span>
                  </Component>
                </label>
              </div>
            </Checkbox>
          </div>
        </Col>
      </div>
    </Row>
  </div>
</DerivationDataScopeFieldOptions>
`;
